aws-lc-rs/src/hkdf.rs

```diff
@@ -349,19 +363,43 @@ impl<L: KeyType> Okm<'_, L> {
         if out.len() != self.len.len() {
             return Err(Unspecified);
         }
-        unsafe {
-            if 1 != HKDF_expand(
-                out.as_mut_ptr(),
-                out.len(),
-                *digest::match_digest_type(&self.prk.algorithm.0.digest_algorithm().id),
-                self.prk.key_bytes.as_ptr(),
-                self.prk.key_len,
-                self.info_bytes.as_ptr(),
-                self.info_len,
-            ) {
-                return Err(Unspecified);
-            };
-        }
+
+        match &self.prk.mode {
```

Comment:
Alternatively, this operation could be implemented by `PrkMode` with the relevant values passed to it. I'm being pedantic, but I like that this way encapsulates the distinction within `PrkMode`.